hub/api/dataset.py

```diff
@@ -357,6 +361,114 @@ def like(
 
         return destination_ds
 
+    @staticmethod
+    def copy(
+        src: str,
+        dest: str,
+        overwrite: bool = False,
+        creds=None,
+        dest_creds=None,
+        token=None,
+        dest_token=None,
+        num_workers: int = 0,
+        scheduler="threaded",
+        progress_bar=True,
+        public: bool = True,
+    ):
+        """Copies dataset at `src` to `dest`.
+
+        Args:
+            src (str): Path to the dataset to be copied.
+            dest (str): Destination path to copy to.
+            overwrite (str): If True and a dataset exists at `destination`, it will be overwritten. Defaults to False.
+            creds (dict, optional): A dictionary containing credentials used to access the dataset at the path.
+                If aws_access_key_id, aws_secret_access_key, aws_session_token are present, these take precedence over credentials present in the environment or in credentials file. Currently only works with s3 paths.
+                It supports 'aws_access_key_id', 'aws_secret_access_key', 'aws_session_token', 'endpoint_url', 'region', 'profile_name' as keys.
+            token (str, optional): Activeloop token, used for fetching credentials to the dataset at `src` if it is a Hub dataset. This is optional, tokens are normally autogenerated.
+            dest_creds (dict, optional): creds required to create / overwrite datasets at destination path. Will use `creds` to copy the dataset if unspecified.
+            dest_token (str, optional): token used to for fetching credentials to destination. Will use `token` if unspecified.
+            num_workers (int): The number of workers to use for copying. Defaults to 0. When set to 0, it will always use serial processing, irrespective of the scheduler.
+            scheduler (str): The scheduler to be used for copying. Supported values include: 'serial', 'threaded', 'processed' and 'ray'.
+                Defaults to 'threaded'.
+            progress_bar (bool): Displays a progress bar if True (default).
+            public (bool, optional): Defines if the dataset will have public access. Applicable only if Hub cloud storage is used and a new Dataset is being created. Defaults to True.
+
+        Returns:
+            Dataset: New dataset object.
+
+        Raises:
+            DatasetHandlerError: If a dataset already exists at destination path and overwrite is False.
+        """
+        if token and not dest_token:
```

Comment:
I don't think we should make these assumptions, the user should specify both separately.